InteractiveHtmlBom/web/ibom.js

```diff
@@ -836,6 +853,37 @@ function prepCheckboxes() {
   }
 }
 
+function populateDarkenWhenCheckedOptions() {
+  var container = document.getElementById("darkenWhenCheckedContainer");
+  container.innerHTML = '';
+  
+  function createOption(name, displayName) {
+    var id = "darkenWhenChecked-" + name;
+
+    var div = document.createElement("div");
+    div.classList.add("radio-container");
+
+    var input = document.createElement("input");
+    input.type = "radio";
+    input.name = "darkenWhenChecked";
+    input.value = name;
+    input.id = id;
+    input.onchange = setDarkenWhenChecked.bind(undefined, name);
+    div.appendChild(input);
+
+    var label = document.createElement("label");
+    label.innerHTML = displayName;
+    label.htmlFor = id;
+    div.appendChild(label);
+
+    container.appendChild(div);
+  }
+  createOption("", "None");
+  for (var checkbox of settings.checkboxes) {
```

Comment:
If there are no checkboxes at all the whole input can be hidden.

Reply:
Done.